hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -331,10 +331,20 @@ writeLog =
 
 -- | Generates a random input for the test by running the provided generator.
 --
-forAll :: (Monad m, Show a, Typeable a, HasCallStack) => Gen m a -> Test m a
-forAll gen = do
+forAll :: (Monad m, Show a, HasCallStack) => Gen m a -> Test m a
+forAll gen =
+  withFrozenCallStack $ forAllWith showPretty gen
+
+-- | Generates a random input for the test by running the provided generator.
+--
+--   /This is a the same as 'forAll' but allows the user to provide a custom/
+--   /rendering function. This is useful for values which don't have a/
+--   /'Show' instance./
+--
+forAllWith :: (Monad m, HasCallStack) => (a -> String) -> Gen m a -> Test m a
+forAllWith render gen = do
```

Comment:
> This is a the same as 'forAll' but allows the user to provide a custom/
> /rendering function. This is useful for values which don't have a/
> /'Show' instance.

Perhaps, *rendering* or *show* could be part of the function name? Perhaps, `forAllWith` is too general? 😕 Hard for me to come up with a good alternative though.